### PR TITLE
BUG: Raise KeyError when indexing a Series with MultiIndex

### LINKED ISSUE #14885: Wrong "Too many indexers" error message when indexing a Series with MultiIndex
#### Code Sample, a copy-pastable example if possible

```python
In [2]: s = pd.Series(range(4), index=pd.MultiIndex.from_product([['a', 'b'], ['c', 'd']]))

In [3]: s.loc['a', 'e']
---------------------------------------------------------------------------
IndexingError                             Traceback (most recent call last)
<ipython-input-3-042ed7ab0463> in <module>()
----> 1 s.loc['a', 'e']

/home/pietro/nobackup/repo/pandas/pandas/core/indexing.py in __getitem__(self, key)
   1308 
   1309         if type(key) is tuple:
-> 1310             return self._getitem_tuple(key)
   1311         else:
   1312             return self._getitem_axis(key, axis=0)

/home/pietro/nobackup/repo/pandas/pandas/core/indexing.py in _getitem_tuple(self, tup)
    799 
    800         # no multi-index, so validate all of the indexers
--> 801         self._has_valid_tuple(tup)
    802 
    803         # ugly hack for GH #836

/home/pietro/nobackup/repo/pandas/pandas/core/indexing.py in _has_valid_tuple(self, key)
    148         for i, k in enumerate(key):
    149             if i >= self.obj.ndim:
--> 150                 raise IndexingError('Too many indexers')
    151             if not self._has_valid_type(k, i):
    152                 raise ValueError("Location based indexing can only have [%s] "

IndexingError: Too many indexers
```

#### Problem description

The raised error, and the message, seem wrong to me.

#### Expected Output
```
KeyError: ('a', 'e')
```
This is what I get if I I take a ``DataFrame`` with the same index and do
```
In [7]: df.loc[('a', 'e'), :] 
```
By the way, ``s.loc['a', 'b']`` (valid key) works just fine, so this is clearly a problem of missing key, and [the docs](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.loc.html) say "``.loc`` will raise a ``KeyError`` when the items are not found."

... and by the way, I would expect _the following_ to raise an ``IndexingError: Too many indexers``:

```
In [12]: s.loc[('a', 'e'), :]
Out[12]: 
a  c    0
   d    1
dtype: int64
```
... instead the tuple is interpreted as a list of labels rather than as a key, and hence it works "fine". Is this behavior desired? (looks a bit inconsistent to me, but I see that it is generalized, i.e. ``DataFrame``s also work this way) If it is, then it is worth fixing the docs where [they mention](url) "A list or array of labels, e.g. ['a', 'b', 'c']." to also mention tuples.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.7.0-1-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.utf8
LOCALE: it_IT.UTF-8

pandas: 0.19.0+196.g5f889a2.dirty
nose: 1.3.7
pip: 8.1.2
setuptools: 28.0.0
Cython: 0.23.4
numpy: 1.11.2
scipy: 0.18.1
statsmodels: 0.8.0.dev0+f80669e
xarray: None
IPython: 5.1.0.dev
sphinx: 1.4.8
patsy: 0.3.0-dev
dateutil: 2.5.3
pytz: 2015.7
blosc: None
bottleneck: 1.2.0dev
tables: 3.2.2
numexpr: 2.6.0
matplotlib: 1.5.3
openpyxl: None
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: None
bs4: 4.5.1
html5lib: 0.999
httplib2: 0.9.1
apiclient: 1.5.2
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.40.0
pandas_datareader: 0.2.1

</details>

### PR DESCRIPTION
**Pull request by ryanreh99**
- [x] closes #14885
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -976,7 +978,12 @@ def _getitem_lowerdim(self, tup):
                 return result
 
         if len(tup) > self.obj.ndim:
-            raise IndexingError("Too many indexers. handle elsewhere")
+            try:
+                self._get_label(tup, axis=self.axis)
+            except KeyError:
```
Comment (jreback) [inline pandas/core/indexing.py]: why would you do this?, pretty redundant code here

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -944,6 +944,8 @@ def _handle_lowerdim_multi_index_axis0(self, tup):
         except TypeError:
             # slices are unhashable
             pass
+        except KeyError:
+            return None
```
Comment (jreback) [inline pandas/core/indexing.py]: you might be able to simply move the len(tup) > self.obj.ndim check right here.

**Review by jreback — CHANGES_REQUESTED** [review]
you additions make this way more complicated, see if you can simplify

File: pandas/core/indexing.py [hunk 3]
```diff
@@ -944,6 +944,9 @@ def _handle_lowerdim_multi_index_axis0(self, tup):
         except TypeError:
             # slices are unhashable
             pass
+        except KeyError as ek:
+            if len(tup) > self.obj.ndim:
```
Comment (jreback) [inline pandas/core/indexing.py]: can you add a comment here on what is happening

File: pandas/core/indexing.py [hunk 4]
```diff
@@ -944,6 +944,9 @@ def _handle_lowerdim_multi_index_axis0(self, tup):
...
+                raise ek
         except Exception as e1:
             if isinstance(tup[0], (slice, Index)):
                 raise IndexingError("Handle elsewhere")
```
Comment (jreback) [inline pandas/core/indexing.py]: is line 978 still needed / hit?

File: pandas/core/indexes/base.py [hunk 5]
```diff
@@ -2874,6 +2874,10 @@ def _convert_scalar_indexer(self, key, kind=None):
 
             elif kind in ['loc'] and is_integer(key):
                 if not self.holds_integer():
+                    try:
```
Comment (jreback) [inline pandas/core/indexes/base.py]: really? what is this actually trying to do. We actually have tests that hit the branches here??

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexing.py (hunk 4 above)
Comment (ryanreh99) [reply pandas/core/indexing.py]: are you referencing this line?
`if len(tup) > self.obj.ndim:
            raise IndexingError("Too many indexers. handle elsewhere")`
I have added the test case where it hits.

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 6]
```diff
@@ -130,6 +131,14 @@ def test_loc_multiindex(self):
             with pytest.raises(KeyError, match=r"^2L?$"):
                 mi_int.ix[2]
 
+        s = Series(range(8), index=MultiIndex.from_product(
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py]: can you make this another test name. add the issue number as a comment

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note in bug fixes; indexing section. the code change looks good.

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 6 above)
Comment (ryanreh99) [reply pandas/tests/indexing/multiindex/test_loc.py]: sorry i dont understand, should i add this to a new function?

File: pandas/core/indexing.py [hunk 7]
```diff
@@ -944,6 +944,12 @@ def _handle_lowerdim_multi_index_axis0(self, tup):
         except TypeError:
             # slices are unhashable
             pass
+        except KeyError as ek:
+            # raise KeyError if number of indexers match
+            # else IndexingError will be raised
+            if (len(tup) == len(self.obj.index._levels)
```
Comment (ryanreh99) [inline pandas/core/indexing.py]: I was wondering should i keep this as it is?
Or should i change `len(self.obj.index._levels)` to `self.obj.index.nlevels`.

Comment (jreback) [reply pandas/core/indexing.py]: yes pls change this

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 6 above)
Comment (jreback) [reply pandas/tests/indexing/multiindex/test_loc.py]: can you make the name more informative, maybe

test_loc_multiindex_too_many_dims

**Review by jreback — CHANGES_REQUESTED** [review]
small comments, ping on green.

**Comment by ryanreh99** [comment]
hello @jreback ping

File: doc/source/whatsnew/v0.25.0.rst [hunk 8]
```diff
@@ -327,6 +327,7 @@ Indexing
 ^^^^^^^^
 
 - Improved exception message when calling :meth:`DataFrame.iloc` with a list of non-numeric objects (:issue:`25753`).
+- Bug in :meth:`DataFrame.loc` and :meth:`Series.loc` where KeyError was not raised for a ``MultiIndex``. Now, IndexingError takes precedence over KeyError (:issue:`14885`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: use double backticks on KeyError; use ``:class:`MultiIndex` ``;  maybe something more like

Bug in :meth:`DataFrame.loc` and :meth:`Series.loc` where ``KeyError`` was not raised for a ``MultiIndex`` when the key was longer that the number of levels in the :class:MUutiIndex.  (:issue:`14885`).

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 8 above)
Comment (ryanreh99) [reply doc/source/whatsnew/v0.25.0.rst]: @jreback actually when key is longer than the levels of the multiindex `IndexingError: Too many indexers` was raised and  is still raised. So would this be more proper?

Bug in :meth:\`DataFrame.loc\` and :meth:\`Series.loc\` where \`\`KeyError\`\` was not raised for a \`\`MultiIndex\`\` when the key was less than or equal to the number of levels in the :class:\`MultiIndex\` (:issue:\`14885\`).

**Comment by ryanreh99** [comment]
made a change to the if statement where a `KeyError` wasn't raised. Added that test.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ryanreh99 very nice

**Comment by ryanreh99** [comment]
@jreback thanks for all the help :D

**Pull request merged**